glide-core/src/cluster_scan_container.rs

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+use logger_core::log_debug;
+use nanoid::nanoid;
+use once_cell::sync::Lazy;
+use redis::{RedisResult, ScanStateRC};
+use std::{collections::HashMap, sync::Mutex};
+
+// This is a container for storing the cursor of a cluster scan.
+// The cursor for a cluster scan is a ref to the actual ScanState struct in redis-rs.
+// In order to avoid dropping it when it is passed between layers of the application,
+// we store it in this container and only pass the id of the cursor.
+// The cursor is stored in the container and can be retrieved using the id.
+// In wrapper layer we wrap the id in an object, which, when dropped, trigger the removal of the cursor from the container.
+// When the ref is removed from the container, the actual ScanState struct is dropped by Rust GC.
```

Comment:
Not that familiar with Rust -- does this happen immediately or do you have to wait for the GC? If the latter, is there a removal callback that could be used to make this deterministic?

Reply:
The idea of Rust is from one side to be very efficient, low-level efficiency on the one side, and on the other side to be very memory efficient. 
That's why you have to deal with the borrow checker and have a hard life with the compiler. 
The idea is that when you pass the compiler; you are good to go. So yes, it is works immediately (of course not on the nano seconds, but very fast).
Anyway, we have no way to trigger something on the internal object since it's part of the redis-rs library, which encapsulates the actual object and we can't directly call to anything on it. Just the API its provide, and the deal is not one of them. Since it is not something that needed if you are using the redis-rs client.
Let's have a one-on-one meeting to discuss it and save long messages here.